src/converters/table-post-fixer.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module table/converters/table-post-fixer
+ */
+
+import Position from '@ckeditor/ckeditor5-engine/src/model/position';
+import { getParentTable, updateNumericAttribute } from './../commands/utils';
+import TableWalker from './../tablewalker';
+
+/**
+ * Injects table post-fixer to the model.
+ *
+ * The role of the tables post-fixer is to ensure that the table's rows has correct structure
+ * after a {@link module:engine/model/model~Model#change `change()`} block was executed.
+ *
+ * The correct structure means that:
+ *
+ * * All table rows have the same width.
+ * * None of a table cells extends vertically beyond its section (either header or body).
+ *
+ * If the table structure is not correct, the post-fixer will automatically correct it in two steps:
+ *
+ * 1. It will clip table cells that extends beyond it section.
+ * 2. It will add empty table cells to those rows which are narrower then the widest table row.
+ *
+ * ## Clipping overlapping table cells
+ *
+ * Such situation may occur when pasting a table (or part of a table) to the editor from external sources.
+ *
+ * As an example see below table which has a table cell (FOO) with a `rowspan` attribute of `2`:
+ *
+ *		<table headingRows="1">
+ *			<tableRow>
+ *				<tableCell rowspan="2">FOO</tableCell>
+ *				<tableCell colspan="2">BAR</tableCell>
+ *			</tableRow>
+ *			<tableRow>
+ *				<tableCell>BAZ</tableCell>
+ *				<tableCell>XYZ</tableCell>
+ *			</tableRow>
+ *		</table>
+ *
+ * will be rendered in the view as:
+ *
+ *		<table>
+ *			<thead>
+ *				<tr>
+ *					<td rowspan="2">FOO</td>
+ *					<td colspan="2">BAR</td>
+ *				</tr>
+ *			</thead>
+ *			<tbody>
+ *				<tr>
+ *					<td>BAZ<td>
+ *					<td>XYZ<td>
+ *				</tr>
+ *			</tbody>
+ *		</table>
+ *
+ * In above example the table will be rendered as a table with two rows - one in the header and second one in the body.
+ * The table cell with FOO contents will not expand to the body section so its `rowspan` attribute will be changed to `1` (and as
+ * the value `1` is a default value the `rowspan` attribute will be removed from the model).
+ *
+ * The table cell with BAZ contents will be in the first column of a table.
+ *
+ * ## Adding missing table cells
+ *
+ * The table post-fixer will insert empty table cells to equalize table rows lengths. The width of a table row is calculated by counting
+ * widths of table cells and widths of columns spanned by cells from rows above in a given row.
```

Comment:
> width of a table row

I'm not sure "width" is the best term here. It comes with a strong flavor of pixels or whatever measurement units.  In fact, you used "length" in the previous sentence :P

> from preceding rows

?